### PR TITLE
Second part

### PR DESCRIPTION
**Pull request by franclobo**
### To-Do List: Add & Remove.

- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.
- By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should be set to 1).
- All changes to the To-Do List should be saved in local storage.

### CODE REVIEW

File: src/modules/class.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    this.completed = completed;
+    this.index = index;
+  }
+}
+
+export const toDoTasks = new ToDoTasks();
+
+export const updateLocalStorage = () => {
+  const newLists = document.querySelectorAll('li');
+  const list = [];
```
Comment (FrederickMih) [inline src/modules/class.js:15]: - [x] I discovered that the page must be refreshed after adding in the first Item in order to add the next items to the list. Please make sure to fix this issue.
- [x] Items added to the list get disappeared after refreshing the page. Please let's make sure that all todos are saved into the local storage correctly. From the requirements of this project, all changes to the To-Do List should be saved in local storage.

File: src/index.js [hunk 2]
```diff
@@ -1,65 +1,55 @@
...
 
+  const editBtnEl = document.createElement('div');
+  editBtnEl.innerHTML = `
+  <i class="fas fa-solid fa-ellipsis-vertical"></i>
+  `;
+  newList.appendChild(editBtnEl);
+
   checkBtnEl.addEventListener('click', () => {
     newList.classList.toggle('checked');
     updateLocalStorage();
```
Comment (FrederickMih) [inline src/index.js:55]: - [x] After editing a task, the updated task is not saved in the local storage. All changes to the To-Do List should be saved in the local storage. Please let's fix this issue.

- [x] Also make sure that deleting a task should update all the remaining item's indexes so they represent the current list order and are unique. In the end, all changes should be saved in the local storage.

File: src/modules/class.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  newLists.forEach((newList) => {
+    list.push({
+      description: toDoTasks.description,
+      completed: newList.classList.contains('checked'),
+      index: list.length,
```
Comment (FrederickMih) [inline src/modules/class.js]: - [x] From the requirements of this project, if the 2nd task is added to the list, the index of that task should be equal to 2. This is not the same behavior in your app as you can see in the screenshot below:

Screenshot from 2022-05-11 00-30-23

Please kindly fix this issue :pray:

File: src/index.js [hunk 4]
```diff
@@ -77,18 +67,26 @@ const toDoList = (task) => {
...
+      taskInput.removeAttribute('readonly');
+      taskInput.focus();
+    } else {
+      taskInput.removeAttribute('readonly', 'readonly');
+      toDoTasks.description = taskInput.value;
+      updateLocalStorage();
+    }
+  });
 
   updateLocalStorage();
```
Comment (FrederickMih) [inline src/index.js:81]: - [x] Once more make sure that all changes to the To-Do List should be saved in local storage.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @franclobo , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



# To Highlights

:fast_forward: GitHub workflow has been followed :heavy_check_mark: 
:fast_forward: Linters has been correctly setup and all checks passed :heavy_check_mark: 
:fast_forward: PR has a good title and summary :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/modules/class.js (hunk 1 above)
Comment (franclobo) [reply src/modules/class.js:15]: Fixed

File: src/index.js (hunk 2 above)
Comment (franclobo) [reply src/index.js:55]: Done.

File: src/modules/class.js (hunk 3 above)
Comment (franclobo) [reply src/modules/class.js]: Fixed

File: src/index.js (hunk 4 above)
Comment (franclobo) [reply src/index.js:81]: updateLocalStorage fixed.

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**